stuff/cron/testing/test_plagiarism_detector.py

```diff
@@ -250,39 +240,64 @@ def test_plagiarism_detector(dbconn: lib.db.Connection) -> None:
-    end_sub_id = start_sub_id + no_of_problems * no_of_users
-    bad_sub_id = [
-        start_sub_id + 2 * no_of_problems,
-        start_sub_id + 2 * no_of_problems + 1
+        cur.execute(
+            '''
+            SELECT `submission_id_1`,
+                    `submission_id_2`
+            FROM `Plagiarisms`
+            WHERE `contest_id` =  %s; 
+                    ''', (contest_id, ))
+        plagiarized_matches = cur.fetchall()
+        plagiarized_submission_ids = set(
+            itertools.chain.from_iterable(
+                (submission_id1, submission_id2)
+                for (submission_id1, submission_id2) in plagiarized_matches))
+
+    # hardcoded
+    bad_submission_ids: List[int] = [
+        first_submission_id + 6, first_submission_id + 7
     ]
-
-    good_sub_ids: Set[Tuple[int, int]] = set()  # expected submission_ids
-    bad_sub_ids: Set[Tuple[int, int]] = set()  # not expected submission_ids
-
-    for sub_id_1 in range(start_sub_id, end_sub_id - no_of_problems):
-        for sub_id_2 in range(sub_id_1 + no_of_problems, end_sub_id,
-                              no_of_problems):
-            if sub_id_2 not in bad_sub_id:
-                if (abs(sub_id_1 - sub_id_2) % no_of_problems == 0):
-                    good_sub_ids.add((sub_id_1, sub_id_2), )
-                    good_sub_ids.add((sub_id_2, sub_id_1), )
-                else:
-                    bad_sub_ids.add((sub_id_1, sub_id_2), )
-                    bad_sub_ids.add((sub_id_2, sub_id_1), )
-
-    for p in plag:
-        assert p in good_sub_ids
-        assert p not in bad_sub_ids
+    good_submission_ids: List[int] = []
+
+    for submission_id in range(first_submission_id, first_submission_id + 9):
+        if (submission_id not in bad_submission_ids):
+            good_submission_ids.append(submission_id)
+
+    for submission_id in good_submission_ids:
+        assert submission_id in plagiarized_submission_ids
+    for submission_id in bad_submission_ids:
+        assert submission_id not in plagiarized_submission_ids
+
+    # hardcoded
+    good_matches: List[Tuple[int, int]] = [
+        (first_submission_id, first_submission_id + 3),
+        (first_submission_id + 1, first_submission_id + 4),
+        (first_submission_id + 2, first_submission_id + 5),
+        (first_submission_id + 2, first_submission_id + 8),
+        (first_submission_id + 5, first_submission_id + 8),
+    ]
+    bad_matches: List[Tuple[int, int]] = []
+    for submission_id1 in range(first_submission_id, first_submission_id + 9):
+        for submission_id2 in range(submission_id1 + 1,
+                                    first_submission_id + 9):
+            if (submission_id1, submission_id2) not in good_matches:
+                bad_matches.append((submission_id1, submission_id2))
+
+    for match in good_matches:
+        assert match in plagiarized_matches
+    for match in bad_matches:
+        assert match not in plagiarized_matches
```

Comment:
```suggestion
    # TODO: Convert the graph of plagiarized submissions into disjoint sets and
    # check groups of plagiarized submissions instead of only whether a submission
    # was plagiarized or not.
    assert plagiarized_matches == set((
      submission_ids[f'{0:032x}'],
      submission_ids[f'{1:032x}'],
      submission_ids[f'{2:032x}'],
      submission_ids[f'{3:032x}'],
      submission_ids[f'{4:032x}'],
      submission_ids[f'{5:032x}'],
      submission_ids[f'{8:032x}'],
    ))
```

where `submission_ids` is a mapping from GUIDs to submission_ids.

Reply:
Done.